File: GoWebApi/app/models/user_sex.go
package models

// UserSex 用户性别
// 使用定义类型而非别名, 避免与普通 int 混用导致赋值越界
type UserSex int

const (
	UserSexUnknown UserSex = 0 // 未填写
	UserSexMale    UserSex = 1 // 男
	UserSexFemale  UserSex = 2 // 女
)

// IsUserSexValid 判断性别值是否有效（仅男/女为有效值）
func IsUserSexValid(s UserSex) bool {
	switch s {
	case UserSexMale, UserSexFemale:
		return true
	}
	return false
}

// GetUserSexText 获取性别中文描述
func GetUserSexText(s UserSex) string {
	switch s {
	case UserSexMale:
		return "男"
	case UserSexFemale:
		return "女"
	default:
		return ""
	}
}

File: RsWebApi/src/app/models/user_sex.rs
//! 用户性别 (对应 Go 项目 app/models/user_sex.go)
//!
//! 以下函数当前业务未使用, 保留以与 Go 版代码结构对齐 (Go 端同样提供这些辅助函数);
//! 由 `#![allow(dead_code)]` 抑制编译告警。
#![allow(dead_code)]

/// 未填写
pub const USER_SEX_UNKNOWN: i32 = 0;
/// 男
pub const USER_SEX_MALE: i32 = 1;
/// 女
pub const USER_SEX_FEMALE: i32 = 2;

/// 判断性别值是否有效（仅男/女为有效值）
pub fn is_user_sex_valid(s: i32) -> bool {
    matches!(s, USER_SEX_MALE | USER_SEX_FEMALE)
}

/// 获取性别中文描述
pub fn get_user_sex_text(s: i32) -> &'static str {
    match s {
        USER_SEX_MALE => "男",
        USER_SEX_FEMALE => "女",
        _ => "",
    }
}
